### PR TITLE
ref(dynamic-sampling): Remove multiplexer

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR aims at removing the `DynamicSamplingFeatureMultiplexer` for simplicity and coherence.

### CODE REVIEW

File: src/sentry/relay/config/__init__.py [hunk 1]
```diff
@@ -41,6 +40,7 @@
 from sentry.utils.http import get_origins
 from sentry.utils.options import sample_modulo
 
+from ...dynamic_sampling.utils import is_on_dynamic_sampling
```
Comment (andriisoldatenko) [inline src/sentry/relay/config/__init__.py:43]: you should use full import path. E.g. sentry.dynamic_sampling.utils

Comment (iambriccardo) [reply src/sentry/relay/config/__init__.py:43]: I think Pycharm did this, will fix! Thanks

File: src/sentry/relay/config/__init__.py [hunk 2]
```diff
@@ -159,9 +159,7 @@ def get_project_config(
 
 
 def get_dynamic_sampling_config(project: Project) -> Optional[Mapping[str, Any]]:
-    feature_multiplexer = DynamicSamplingFeatureMultiplexer(project)
-
-    if feature_multiplexer.is_on_dynamic_sampling:
+    if is_on_dynamic_sampling(project):
```
Comment (andriisoldatenko) [inline src/sentry/relay/config/__init__.py:162]: The idea is to use directly: like we do across all sentry code base:
```
 features.has("organizations:dynamic-sampling", project.organization) and options.get(
        "dynamic-sampling:enabled-biases"
    )
```

Comment (andriisoldatenko) [reply src/sentry/relay/config/__init__.py:162]: just grep by: `features.has("organizations:`

Comment (iambriccardo) [reply src/sentry/relay/config/__init__.py:162]: I didn't like this approach as soon as I saw it used throughout the codebase, but if we you think that we should unify, then let's do that

Comment (andriisoldatenko) [reply src/sentry/relay/config/__init__.py:162]: python -c "import this" | rg exp
4:Explicit is better than implicit.

🤷

Comment (iambriccardo) [reply src/sentry/relay/config/__init__.py:162]: I think that a proper design of the features would be to allow for composite ones but that is another story.

Comment (andriisoldatenko) [reply src/sentry/relay/config/__init__.py:162]: feature flagger already supports it.

Comment (iambriccardo) [reply src/sentry/relay/config/__init__.py:162]: I meant a composite that would span different contexts and for example automatically get access to options, that would be quite nice

File: src/sentry/api/endpoints/project_details.py [hunk 3]
```diff
@@ -621,99 +617,108 @@ def put(self, request: Request, project) -> Response:
                 ]
         # TODO(dcramer): rewrite options to use standard API config
         if has_project_write:
-            options = request.data.get("options", {})
-            if "sentry:origins" in options:
+            request_options = request.data.get("options", {})
```
Comment (andriisoldatenko) [inline src/sentry/api/endpoints/project_details.py:620]: please remove this unnecessary change, and try to refactor it separately.

Comment (iambriccardo) [reply src/sentry/api/endpoints/project_details.py:620]: This change is necessary because of shadowing, should I just change the name of the import from `options` to `sentry_options`?

Comment (andriisoldatenko) [reply src/sentry/api/endpoints/project_details.py:620]: oh i see, yeah probobly `from sentry import options as sentry_options`

**Review by andriisoldatenko — APPROVED** [review]
